vendor/brave-ios/BUILD.gn

```diff
@@ -69,6 +69,7 @@ ios_framework_bundle("brave_rewards_ios_framework") {
   deps = [
     "//brave/vendor/bat-native-ledger",
     "//brave/vendor/bat-native-ads",
+    "//net:net",
```

Comment:
?

Reply:
For ios:

```
09:07:40  > brave@1.13.21 gn_check /Users/jenkins/jenkins/workspace/pr-brave-browser-gn_check
09:07:40  > node ./scripts/commands.js gn_check "Release" "--target_os=ios"
09:07:40  
09:07:40  Recursing through GRD to find GRDP files...
09:07:40  Done recursing through GRD to find GRDP files.
09:07:40  using git cache path /Users/jenkins/cache
09:07:40  -------------------------------
09:07:40  /Users/jenkins/jenkins/workspace/pr-brave-browser-gn_check/src
09:07:40  > gn check /Users/jenkins/jenkins/workspace/pr-brave-browser-gn_check/src/out/ios_Release //brave/*
09:07:44  ERROR at //brave/vendor/brave-ios/Ledger/BATBraveLedger.mm:31:10: Can't include this header from here.
09:07:44  #import "net/base/registry_controlled_domains/registry_controlled_domain.h"
09:07:44           ^----------------------------------------------------------------
09:07:44  The target:
09:07:44    //brave/vendor/brave-ios:brave_rewards_ios_framework_arch_shared_library_sources
09:07:44  is including a file from the target:
09:07:44    //net:net
09:07:44  
```

Reply:
Yeah we were piggybacking on ledger/ads usage rather than import it ourself

Reply:
@kylehickinson @tmancey it would be better if that was handled inside ledger now that we allow chromium deps there, but sounds like this is needed for now